theories/crypto/PKE.ec

```diff
@@ -163,3 +164,71 @@ module Correctness (S:Scheme) = {
     return (m' = Some m);
   }
 }.
+
+module type CAdversary = {
+   proc find(pk : pkey, sk : skey) : plaintext 
+}.
+
+module CorrectnessAdv(S : Scheme, A : CAdversary) = {
+  proc main() : bool = {
+    var pk, sk, c, m, m';
+    (pk,sk) <@ S.kg();
+    m <@ A.find(pk,sk);
+    c <@ S.enc(pk, m);
+    m' <@ S.dec(sk,c);
+
+    return m' = Some m;
+  }
+}.
+
+(* Extensions to ROM *)
+
+theory PKE_ROM.
+
+clone import ROM as RO.
+
+module type SchemeRO(H : POracle) = {
+  include Scheme
+}.
+
+module type AdversaryRO(H : POracle) = {
+  include Adversary
+}.
+
+module type CAdversaryRO(H : POracle) = {
+  include CAdversary
+}.
+
+module type CPAGame(S: Scheme, A : Adversary) = {
+   proc main() : bool
+}.
+
+module CPAGameROM(G : CPAGame, S : SchemeRO, A : AdversaryRO, O : Oracle) = {
```

Comment:
See commit comment: do we want to curry functor parameters?

Reply:
Not sure. You mean setting RO as the first parameter?

Reply:
I'm just going to accept it and leave making a decision on the syntax of module params for Future Me. He already hates me anyway.